app/src/handlers/census/events/ContinentUnlockEvent.ts

```diff
@@ -0,0 +1,59 @@
+import { injectable } from 'inversify';
+import { GenericEvent } from '../../../types/censusEventTypes';
+import IllegalArgumentException from '../../../exceptions/IllegalArgumentException';
+import Parser from '../../../utils/parser';
+import { Zone } from '../../../constants/zone';
+import FactionUtils from '../../../utils/FactionUtils';
+import { Faction } from '../../../constants/faction';
+import ZoneUtils from '../../../utils/ZoneUtils';
+import { ContinentUnlock } from 'ps2census/dist/client/utils/PS2Events';
+
+@injectable()
+export default class ContinentUnlockEvent {
+    public readonly worldId: number;
+    public readonly zone: Zone;
+    public readonly timestamp: number;
+    public readonly triggeringFaction: Faction;
+    public readonly vsPopulation: number;
+    public readonly ncPopulation: number;
+    public readonly trPopulation: number;
+
+    public constructor(
+        event: GenericEvent
+    ) {
+        // Already fixed in master. Will be fixed with #36
+        const continentUnlockEvent = event as unknown as ContinentUnlock;
+        this.worldId = Parser.parseArgumentAsNumber(continentUnlockEvent.world_id);
+        if (isNaN(this.worldId)) {
+            throw new IllegalArgumentException('world_id');
+        }
+        // No need to check, ZoneUtils will validate the argument
+        this.zone = ZoneUtils.parse(Parser.parseArgumentAsNumber(continentUnlockEvent.zone_id));
+        this.timestamp = Parser.parseArgumentAsNumber(continentUnlockEvent.timestamp);
+        if (isNaN(this.timestamp)) {
+            throw new IllegalArgumentException('timestamp');
+        }
+        this.vsPopulation = Parser.parseArgumentAsNumber(continentUnlockEvent.vs_population);
+        if (isNaN(this.vsPopulation)) {
+            throw new IllegalArgumentException('vs_population');
+        }
+        this.ncPopulation = Parser.parseArgumentAsNumber(continentUnlockEvent.nc_population);
+        if (isNaN(this.ncPopulation)) {
+            throw new IllegalArgumentException('nc_population');
+        }
+        this.vsPopulation = Parser.parseArgumentAsNumber(continentUnlockEvent.vs_population);
+        if (isNaN(this.vsPopulation)) {
+            throw new IllegalArgumentException('vs_population');
+        }
+        this.trPopulation = Parser.parseArgumentAsNumber(continentUnlockEvent.tr_population);
+        if (isNaN(this.trPopulation)) {
```

Comment:
Needs origins (check the other event classes too, might have forgot to pick up on this last review)